modules/dnn/src/ocl4dnn/src/math_functions.cpp

```diff
@@ -452,7 +452,7 @@ bool ocl4dnnGEMV<float>(const CBLAS_TRANSPOSE TransA,
         uint row_size = M;
         uint col_size = N;
         size_t localsize[] = { 128 };
-        size_t globalsize[] = { row_size / 4 * localsize[0] };
+        size_t globalsize[] = { std::max(row_size / 4, uint(1)) * localsize[0] };
```

Comment:
My suggestion is to put this kernel call under condition `if (row_size >= 4)`.
`row_size` is not passed to kernel, so there is way to avoid memory corruption.

Reply:
@alalek, Thanks! Edited.